src/Components/Components/src/NavigationManager.cs

```diff
@@ -274,6 +281,128 @@ protected void NotifyLocationChanged(bool isInterceptedLink)
         }
     }
 
+    /// <summary>
+    /// Notifies the registered <see cref="IHandleLocationChanging"/> handlers of the current location change.
+    /// </summary>
+    /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI.</param>
+    /// <param name="isNavigationIntercepted">Whether this navigation was intercepted from a link.</param>
+    /// <param name="forceLoad">Whether the navigation is attempting to bypass client-side routing.</param>
+    /// <returns>A <see cref="ValueTask{TResult}"/> representing the completion of the operation.</returns>
+    public async ValueTask<bool> NotifyLocationChanging(string uri, bool isNavigationIntercepted, bool forceLoad)
```

Comment:
Does this have to be public? I suspect protected might suffice for the platform-specific subclasses to call it.